doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+  For example ``exampleCmdLineTask.ExampleCmdLineConfig`` creates the statistics subtask with name ``stats`` because the config field for that subtask is ``stats = lsst.pex.config.ConfigurableField(...)``.
+
+  Task names are used for the hierarchy of task and subtask metadata.
+  Also, for command-line tasks the name is used as a component of the of the dataset type for saving the task's configuration and metadata; see :ref:`creating-a-command-line-task-persisting-config-and-metadata` for more information.
+
+- ``RunnerClass``: only needed for command-line tasks; see :doc:`creating-a-command-line-task` for more information.
+
+- ``canMultiprocess``: only needed for command-line tasks; see :doc:`creating-a-command-line-task` for more information.
+
+Here are the class variables for ``ExampleCmdLineTask``:
+
+.. code-block:: python
+
+   ConfigClass = ExampleCmdLineConfig
+   _DefaultName = "exampleTask"
+
+.. _creating-a-task-class-methods:
+
+Methods
+=======
+
+Tasks have the following important methods:
+
+- :ref:`__init__ <creating-a-task-class-init-method>`: construct and initialize a task.
+- :ref:`run <creating-a-task-class-run-method>`: process one item of data.
+
+These methods are described in more depth below.
+
+.. _creating-a-task-class-init-method:
+
+The ``__init__`` method
+-----------------------
+
+Use the ``__init__`` method (task constructor) to do the following:
+
+- Call the parent task's ``__init__`` method
+- Make subtasks by calling ``self.makeSubtask(name)``, where ``name`` is the name of a field of type `lsst.pex.config.ConfigurableField` in your :ref:`task's configuration <creating-a-task-configuration>`.
+- Make a schema if your task uses an `lsst.afw.table`.
+  For an example of such a task `lsst.pipe.tasks.calibrate.CalibrateTask`.
+- Initialize any other instance variables your task needs.
+
+Here is ``exampleCmdLineTask.ExampleCmdLineTask.__init__``:
+
+.. code-block:: python
+
+   def __init__(self, *args, **kwargs):
+       """Construct an ExampleCmdLineTask
+       Call the parent class constructor and make the "stats" subtask from the config field of the same name.
+       """
+       pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+       self.makeSubtask("stats")
+
+That task creates a subtask named ``stats`` to compute image statistics.
+Here is the ``__init__`` method for the default version of the `stats` subtask:
+
+exampleTask.ExampleSigmaClippedStatsTask, which is slightly more interesting:
+
+.. code-block:: python
+
+   def __init__(self, *args, **kwargs):
+       """!Construct an ExampleSigmaClippedStatsTask
+       The init method may compute anything that that does not require data.
+       In this case we create a statistics control object using the config
+       (which cannot change once the task is created).
+       """
+       pipeBase.Task.__init__(self, *args, **kwargs)
+       self._badPixelMask = MaskU.getPlaneBitMask(self.config.badMaskPlanes)
```

Comment:
`MaskU` can and should now be written `Mask` (and if you don't like that, `MaskX` will at least run).

I don't quite know how JSick would know that -- how do we keep this sort of thing in sync?  The example as written won't run.